pkgs/development/tools/misc/terraform-plugin-docs/default.nix

```
@@ -0,0 +1,39 @@
+{ lib, buildGoModule, fetchFromGitHub, makeWrapper, musl, terraform, go }:
+
+buildGoModule rec {
+  pname = "terraform-plugin-docs";
+  version = "0.16.0";
+
+  src = fetchFromGitHub {
+    owner = "hashicorp";
+    repo = "terraform-plugin-docs";
+    rev = "v${version}";
+    hash = "sha256-5vbi69GMgkzvN3aEQbNTbk99rg+kfvAvUrdDsuyIm9s=";
+  };
+
+  subPackages = [ "cmd/tfplugindocs" ];
+  vendorHash = "sha256-AjW6BokLVDkIWXToJ7wNq/g19xKTAfpQ/gVlKCV5qw0=";
+  allowGoReference = true;
+  nativeBuildInputs = [ musl makeWrapper ];
+  CGO_ENABLED = 0;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.commit=${src.rev}"
+    "-linkmode external"
+    "-extldflags '-static -L${musl}/lib'"
+  ];
+
+  postInstall = ''
+    wrapProgram $out/bin/tfplugindocs --prefix PATH : ${lib.makeBinPath [ terraform go ]}
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/hashicorp/terraform-plugin-docs";
+    description = "Generate and validate Terraform plugin/provider documentation.";
+    maintainers = with maintainers; [ catouc ];
```

```suggestion
    maintainers = with maintainers; [ catouc ];
    platforms = platforms.linux ++ platforms.darwin;
```

You can also enable darwin support for this package if you want.
Looks like terraform-plugin-docs upstream supports darwin.

https://github.com/hashicorp/terraform-plugin-docs/blob/main/.goreleaser.yml#L14